### ISSUE TITLE
Issue Title

### ISSUE THREAD

**Issue by AdamJStacey** [primefaces/primeng#19292]
### Describe the bug

Issue 1) When using arrow keys to navigate a tree with droppableNodes = true it throws an error and does not work

<img width="883" height="244" alt="Image" src="https://github.com/user-attachments/assets/b1e74b55-8d7a-4778-8aee-a5639ed4dbd8" />

Reproducible at https://primeng.org/tree#dragdrop

this was caused by https://github.com/primefaces/primeng/commit/224b85cfc05ac8f3ecdd7698247e9e2d1689e4e0 as the focus node is expecting 2 children on the element, but the first one got removed in this pr, and so has been an issue since v20.2.0 and still exists in the most recent version

Issue 2) When collapsing a node the focus can get thrown outside of the entire tree which I can consistently reproduce at https://primeng.org/tree#selection

### Pull Request Link

https://github.com/primefaces/primeng/pull/19291

### Reason for not contributing a PR

- [ ] Lack of time
- [ ] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://primeng.org/tree#dragdrop

### Environment

Any

### Angular version

v20+

### PrimeNG version

v21

### Node version

22.14.0

### Steps to reproduce the behavior

For issue one 
1) click any node in a droppableNodes Tree such as on https://primeng.org/tree#dragdrop
2) click any arrow key
3) check the dev console for the error

For issue 2 
I can consistently reproduce this at https://primeng.org/tree#selection
1. click to expand Movies
2. click to expand Al Pacino
3. press the left arrow key to collapse the node
4. press up or down and see the cursor blinking outside of the tree

### Expected behavior

using arrow keys should navigate the tree and not throw any error
